packages/roosterjs-content-model/lib/publicApi/utils/selectWord.ts

```diff
@@ -0,0 +1,103 @@
+import { ContentModelBlockGroup } from '../../publicTypes/group/ContentModelBlockGroup';
+import { ContentModelSegment } from '../../publicTypes/segment/ContentModelSegment';
+import { ContentModelText } from '../../publicTypes/segment/ContentModelText';
+import { createText } from '../../modelApi/creators/createText';
+
+export function selectWord(
+    path: ContentModelBlockGroup[],
+    marker: ContentModelSegment
+): ContentModelSegment[] {
+    const parent = path[0];
+    const block = parent.blocks[0];
```

Comment:
use iterateSelections() to find out the selections. Because parent.blocks[0] may not always be the right block.

Reply:
Which selections do I need to find?

Reply:
iterateSelection() need a callback as parameter. In that callback it will give you all info for a selection that it find. So you just need to check the first selection with a valid paragraph and a selection marker, then return true in the callback to stop it searching:

```ts
iterateSelection([model], (path, tableContext, block, segments) => {
  if (block?.blockType == 'Paragraph' && segments?.length == 1 && segments[0].segmentType == 'SelectionMarker') {
    // Now you have found the selection, block is the parent paragraph, segments[0] is the marker, then you can do your split here
  }

    return true; // this will stop the searching. You can always stop it since when run to here, it must only one single selection
});
```